### PR TITLE
Simplify class hierarchy for ordinals field data.

### PR DESCRIPTION
**Pull request by jtibshirani**
This PR simplifies the hierarchy for ordinals field data classes:
* Remove `AbstractIndexFieldData`, since only `AbstractIndexOrdinalsFieldData`
inherits directly from it.
* Make `SortedSetOrdinalsIndexFieldData` extend
`AbstractIndexOrdinalsFieldData`. This lets us remove some redundant code.

It's probably easiest to review each commit separately -- this makes it clearer
that no logic has changed.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/index/fielddata/plain/SortedSetOrdinalsIndexFieldData.java [hunk 1]
```diff
@@ -78,12 +72,8 @@ public SortedSetOrdinalsIndexFieldData build(
         }
     }
 
-    protected final String fieldName;
-    private final IndexFieldDataCache cache;
-    private final CircuitBreakerService breakerService;
-    private final Function<SortedSetDocValues, ScriptDocValues<?>> scriptFunction;
-    private final ValuesSourceType valuesSourceType;
     private static final Logger logger = LogManager.getLogger(SortedSetOrdinalsIndexFieldData.class);
+    private final Function<SortedSetDocValues, ScriptDocValues<?>> scriptFunction;
```
Comment (nik9000) [inline server/src/main/java/org/elasticsearch/index/fielddata/plain/SortedSetOrdinalsIndexFieldData.java:76]: Does the superclass also have this member?

Comment (jtibshirani) [reply server/src/main/java/org/elasticsearch/index/fielddata/plain/SortedSetOrdinalsIndexFieldData.java:76]: Oops, yes this should only live in the superclass now.

**Comment by jtibshirani** [comment]
This is now ready for another look.

**Review by nik9000 — APPROVED** [review]

**Review by not-napoleon — APPROVED** [review]
LGTM

**Pull request merged**